email_backend/data_management/generate_csv.js

```diff
@@ -0,0 +1,80 @@
+import { connect_db } from "../transpiled_build/db_utils/connect_db";
+import { get_templates } from "../transpiled_build/template_utils";
+import { make_mongoose_model_from_original_template } from "../transpiled_build/db_utils/log_email_and_meta_to_db";
+import _ from "lodash";
+import fs from "fs";
+import mongoose from "mongoose";
+connect_db();
+const templates = get_templates();
+_.map(templates, (template_value, template_name) => {
+  const collection = make_mongoose_model_from_original_template({
+    original_template: template_value,
+    template_name,
+  });
+
+  //if it happens that later on we add another key with a json value,
+  //change reduced json to handle "key.value". For the time being, the CSV
+  //will be cleaner if we flatten the "key.value" to "value"
+
+  collection.find({}, function (err, submissions) {
+    const column_names = _.chain(_.keys(collection.schema.paths))
+      .filter((key) => {
+        return key !== "__v";
+      })
+      .map((key) => {
+        const key_split = key.split(".");
+        return key_split.length > 1 ? key_split[1] : key;
+      })
+      .value();
+
+    const reduced_json = _.map(submissions, function (sub) {
```

Comment:
Variable names are good for the most part, but lose cohesion as they're re-mapped through the file.

I.e.
  - Get documents from DB, name them `submissions` (presumably an array of the logs for submitted emails, the name `submissions` could be clearer about that, e.g. `email_logs`).
  - map `submissions` with elements as `sub` here (can assume short for a singular "submission", but what a "submission" is still requires knowledge from outside this file)
  - reduce the `sub` with elements as `value` (not a useful var name, who knows what a `value` of a `submission` is? Well, we do, it's a generic field for a logged email, could capture that in a var name, e.g. `email_log_field`). Accumulate the reduced `sub` `value`s as `result` (I'll give a pass here, generally tricky to give the accumulator in a reduce a better name than that)
  - store the reduced `submissions` as `reduced_json`. This is extra misleading, because it's A) an array of "reduced json"s, not a single "reduced jsons", b) has no JSON, just JS objects. More like `flattened_email_logs`

Don't mean to be overly critical. Like I said, taken on their own none of them are necessarily bad names, I just wanted to illustrate how they could be chosen to provide more information to future devs. It's extra important to clearly name the inputs, outputs, and internal arguments of any functional pipeline-style code (like lodash), because the names of variables at each step are the best summary of what each step is intended to do.

With all the lodash we write, I think good naming is particularly valuable. If you know what a step takes and what it outputs (and have confidence that there are no side effects), then you can generally skip reading the implementation. Real time saver, and makes debugging other dev's code a whole lot easier.